numpy/lib/tests/test_histograms.py

```diff
@@ -159,6 +160,15 @@ def test_weights(self):
             weights=[2, 1, 1, 1, 1, 1, 1, 1, 1], density=True)
         assert_almost_equal(a, [.2, .1, .1, .075])
 
+    def test_weights_warning(self):
+        a = np.array([1,2,3,4,5,6,7,8,9,10])
+        with warnings.catch_warnings(record=True) as w:
+            warnings.simplefilter("always")
+            hist, bin_edges = np.histogram(a,bins=10, weights=range(len(a)))
```

Comment:
Why should this warn?

Reply:
Want to use `assert_warns` in any case, I think. Any other tests where the warning is raised also need to suppress it.